lib/opensusebasetest.pm

```diff
@@ -780,7 +791,11 @@ sub select_serial_terminal {
             $console = $root ? 'root-sut-serial' : 'sut-serial';
         }
     } elsif ($backend =~ /^(ikvm|ipmi|spvm)$/) {
-        $console = 'root-ssh';
+        if (check_var('SSH_CONSOLE', 0)) {
+            $console = $root ? 'root-console' : 'user-console';
+        } else {
+            $console = 'root-ssh';
+        }
```

Comment:
this looks nice however the original idea of consoles was that this can be covered by them directly as well, so we want to write e.g. `select_console 'root-console'` and internally the decision is made, e.g. if it's a getty on a tty or a ssh session or a serial virtio terminal. Take a look into susedistribution.pm in the methods `init_consoles` as well as `activate_console` to get an idea what is done for other consoles.

Reply:
Isn't it that we have multiple consoles at the same time and one has advantages over the other.  I thinking on a scenario where I need a fast text base console, so I use virtio_terminal. But when I need to reboot I defiantly need something different as virtio_terminal doesn't give me the needed output.

Would it mean, that I remove/add `root-console`?
What if I need two consoles to do some things in parallel?

Reply:
> this looks nice however the original idea of consoles was that this can be covered by them

@okurz "original idea": since when? Before `select_serial_terminal()` or even before virtio?

> directly as well, so we want to write e.g. `select_console 'root-console'` and internally the decision is made, e.g. if it's a getty on a tty or a ssh session or a serial virtio terminal. Take a look into susedistribution.pm in the methods `init_consoles` as well as `activate_console` to get an idea what is done for other consoles.

OT: mixing functionality of passing specific console parameter and ignore it in some cases looks confusing. I guess it's better to have lowlevel function `select_console $console` which always tries to select specific console and some smart wrapper (which should be used in most cases, but there is no good default for anybody). That's what people suggested when I was introducing `select_serial_terminal()`.  If there is such functionality on `select_console`, than it should be for calling without console parameter.

To get to virtio/serial terminal you need to have variables. As there are both `root-console` and virtio/serial terminal => IMHO you still need to have variables to detect whether you want to have virtio/serial terminal or `root-console` as there is no good default. e.g ou can still have needles test on s390x, so you cannot use `root-sut-serial`. Or am I missing something?

Default could be if we were able to detect using needles in tests. But even then we'd be forcing users to use fast virtio/serial console if they don't needles, which I'm not sure it's a good idea.

`select_serial_terminal()` allows not forcing anybody to any default as there is still `select 'root-console'`. The functionality of `select_serial_terminal()` could be moved to `select_console()`, but than many tests would have to setup variables (there is no good default thus it cannot be defined on medium).